lib/toopher-web/OAuth.php

```diff
@@ -481,16 +481,18 @@ public function build_signature($signature_method, $consumer, $token) {
   private static function generate_timestamp() {
     return time();
   }
-
+    
   /**
-   * util function: current nonce
+   * util function: generate secure nonce
    */
-  private static function generate_nonce() {
-    $mt = microtime();
-    $rand = mt_rand();
-
-    return md5($mt . $rand); // md5s look nicer than numbers
+  private static function generate_nonce($size = 16) {
+    if (!function_exists('openssl_random_pseudo_bytes')) {
+      throw new \InvalidArgumentException('OpenSSL not supported.');
```

Comment:
Can we please reword this to target the user's needs instead of the system's state (e.g. "Missing required dependency: OpenSSL")?